applications/ConstitutiveLawsApplication/custom_constitutive/associative_plastic_damage_model.cpp

```diff
@@ -0,0 +1,1092 @@
+    )
+{
+    double det = 0.0;
+    MathUtils<double>::InvertMatrix(rPDParameters.ComplianceMatrix,
+        rPDParameters.ConstitutiveMatrix, det);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TYieldSurfaceType>
+void AssociativePlasticDamageModel<TYieldSurfaceType>::UpdateInternalVariables(
+    PlasticDamageParameters &rPDParameters
+    )
+{
+    mPlasticDissipation        = rPDParameters.PlasticDissipation;
+    mDamageDissipation         = rPDParameters.DamageDissipation;
+    mThreshold                 = rPDParameters.Threshold;
+    noalias(mPlasticStrain)    = rPDParameters.PlasticStrain;
+    noalias(mComplianceMatrix) = rPDParameters.ComplianceMatrix;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TYieldSurfaceType>
+void AssociativePlasticDamageModel<TYieldSurfaceType>::CheckMinimumFractureEnergy(
+    ConstitutiveLaw::Parameters& rValues,
+    PlasticDamageParameters &rPDParameters
+    )
+{
+    const Properties& r_material_properties = rValues.GetMaterialProperties();
+    const bool is_yield_symmetric = r_material_properties.Has(YIELD_STRESS_TENSION) ? false : true;
+
+    const double young_modulus = r_material_properties[YOUNG_MODULUS];
+    const double yield = (is_yield_symmetric == false) ? r_material_properties[YIELD_STRESS_TENSION] :
+        r_material_properties[YIELD_STRESS];
+    const double fracture_energy = r_material_properties[FRACTURE_ENERGY];
+
+    const double hlim = 2.0 * young_modulus * fracture_energy / (std::pow(yield, 2));
+    KRATOS_ERROR_IF(rPDParameters.CharacteristicLength > hlim) << "The Fracture Energy is to low: " <<
+        rPDParameters.CharacteristicLength << std::endl;
+
+    if (is_yield_symmetric == false) { // Check frac energy in compression
+        const double yield_compression =  r_material_properties[YIELD_STRESS_COMPRESSION];
+        const double fracture_energy_compr = r_material_properties[FRACTURE_ENERGY_COMPRESSION];
+        const double hlim_compr = 2.0 * young_modulus * fracture_energy_compr / (std::pow(yield_compression, 2));
+        KRATOS_ERROR_IF(rPDParameters.CharacteristicLength > hlim_compr) << "The Fracture Energy in compression is to low: " <<
+            rPDParameters.CharacteristicLength << std::endl;
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TYieldSurfaceType>
+void AssociativePlasticDamageModel<TYieldSurfaceType>::CalculateElasticComplianceMatrix(
+    BoundedMatrixType& rC,
+    const Properties& rMaterialProperties
+    )
+{
+    const double E = rMaterialProperties[YOUNG_MODULUS];
```

Comment:
Is this the isotropic constitutive tensor? (should be in utilities to reduce code duplciation, just saying...)

Reply:
Not exactly Vicente! this methos computes the Compliance Elastic matrix (inverse of the constitutive matrix), I think that this is only used in here

Reply:
OK, just to know